modules/core/src/main/scala/doobie/util/transactor.scala

```diff
@@ -361,13 +362,13 @@ object transactor  {
        * @param driver     the class name of the JDBC driver, like "org.h2.Driver"
        * @param url        a connection URL, specific to your driver
        */
-      def apply[M[_]: Async: ContextShift](
+      def apply[M[_]: Async: ContextShift: DefaultMessageLogger](
         driver: String,
         url:    String
       ): Transactor.Aux[M, Unit] =
         apply(driver, url, defaultBlocker)
 
-      def apply[M[_]: Async: ContextShift](
+      def apply[M[_]: Async: ContextShift: DefaultMessageLogger](
```

Comment:
Do you think it'll be better to have this as a default parameter? It feels more explicit and approachable - the user doesn't need to know anything about implicit prioritization to understand how to use it.

Reply:
It can't be a default parameter because need `Sync[M]` to construct an instance (you can't refer to a prior argument in a default value). `DefaultMessageLogger` is a hack to get around this. All users need to do is bring a logger into scope where the transactor is constructed. Doesn't seem that hard to me.

Reply:
true we'll cover it with documentation